### PR TITLE
feat(ui): integrate UI with backend service

### PR DESCRIPTION
**Pull request by adglkh**
This change
1. enables UI to integrate with backend service by talking to the
   custom gateway connector.
2. flake8 sanitation for python scripts(backend).
3. drop the useless classes/components that is used to create
   the default AnboxStreamGatewayConnector defined in js sdk

Test: 1. create config.yaml which contains the stream gateway URL and API
         token
      2. export
      3. launch the backend service and navigate to https://0.0.0.0:8002
      4. check the list of application, which matches the existing
         applications installed on ams node.
      5. start a game and check if the streaming works.

### CODE REVIEW

File: ui/lib/api/application.dart [hunk 1]
```diff
@@ -14,9 +14,23 @@
 // limitations under the License.
 //
 
+import 'dart:io';
+
 class Application {
+  static Map<String, String> appDesMap = const {
+    'Beach Buggy Racing 2': 'Join the Beach Buggy Racing League and compete against drivers and cars from around the world. Race through Egyptian pyramids, dragon-infested castles, pirate ship wrecks, and experimental alien bio-labs. Collect and upgrade an arsenal of fun and wacky Powerups. Recruit new drivers, assemble a garage full of cars and race your way to the top of the League.',
```
[the remark this answers is not in the document]
Comment (adglkh) [reply ui/lib/api/application.dart:21]: There is no description for bombsquad yet. I'll add it later along with other games which are going to be installed via the installation script.

[the remark this answers is not in the document]
Comment (morphis) [reply ui/lib/api/application.dart:21]: Sounds good, thanks

File: ui/lib/homepage.dart [hunk 2]
```diff
@@ -46,45 +70,81 @@ class Homepage extends StatelessWidget {
...
+                        '${snapshot.error}',
+                        textAlign: TextAlign.center,
+                        style: TextStyle(
+                          fontSize: 26,
+                          color: Colors.white,
+                        )
+                      )
+                    );
+                }
+                return const CircularProgressIndicator();
```
Comment (morphis) [inline ui/lib/homepage.dart:142]: The indicator appears on the top of the page but we should center it instead

File: ui/lib/homepage.dart [hunk 3]
```diff
@@ -46,45 +70,81 @@ class Homepage extends StatelessWidget {
...
+                    constraints: const BoxConstraints(maxWidth: 1100),
+                    alignment: Alignment.center,
+                    child: Column(
                       children: [
+                        FeaturedApp(
+                          id: apps[0].id,
+                          name: apps[0].name,
+                          backgroundUrl: apps[0].background,
+                          description: apps[0].description,
+                          onPlay: widget.onPlay
```
Comment (morphis) [inline ui/lib/homepage.dart:104]: You haven't fixed stopping the stream when leaving fullscreen yet, right?

File: ui/lib/homepage.dart [hunk 4]
```diff
@@ -46,45 +70,81 @@ class Homepage extends StatelessWidget {
...
+                        ),
                         Container(
-                          margin: const EdgeInsets.all(20),
-                          child: const Text(
-                            'Available applications',
```
Comment (morphis) [inline ui/lib/homepage.dart:70]:
```suggestion
Available games
```

**Review by morphis — CHANGES_REQUESTED** [review]

File: ui/lib/homepage.dart (hunk 2 above)
Comment (adglkh) [reply ui/lib/homepage.dart:147]: Fixed.

File: ui/lib/homepage.dart [hunk 5]
```diff
@@ -13,24 +13,48 @@
...
-class Homepage extends StatelessWidget {
-  final Function() onPlay;
+Future<List<Application>> fetchApps(List<String> ids) async {
+  final response = await http
+      .get(Uri.parse(Uri.base.origin.toString() + "/1.0/games"));
+  if (response.statusCode == 200) {
+    List<String> applist = jsonDecode(response.body).where((i) => ids.contains(i));
+    return List<Application>.from(applist.map((v) => Application.fromString(v)));
+  } else {
+    throw Exception('Failed to load applications');
```
[the remark this answers is not in the document]
Comment (adglkh) [reply ui/lib/homepage.dart]: Fixed.

File: ui/lib/homepage.dart (hunk 3 above)
Comment (adglkh) [reply ui/lib/homepage.dart:109]: Fixed.

File: ui/lib/homepage.dart (hunk 4 above)
Comment (adglkh) [reply ui/lib/homepage.dart:70]: Fixed.

**Review by morphis — APPROVED** [review]
LGTM

**Pull request merged**